src/main/java/de/dennisguse/opentracks/AbstractTrackDeleteActivity.java

```diff
@@ -38,21 +37,46 @@
  *
  * @author Jimmy Shih
  */
-//TODO It is actually not a List; we need a better name for this class
 //TODO Check if this class is still such a good idea; inheritance might limit maintainability
-public abstract class AbstractListActivity extends AbstractActivity implements ConfirmDeleteCaller {
+public abstract class AbstractTrackDeleteActivity extends AbstractActivity implements ConfirmDeleteCaller {
 
-    private static final String TAG = AbstractListActivity.class.getSimpleName();
+    private static final String TAG = AbstractTrackDeleteActivity.class.getSimpleName();
 
     protected static final int GPS_REQUEST_CODE = 6;
-    private static final int DELETE_REQUEST_CODE = 3;
+
+    private TrackDeleteServiceConnection trackDeleteServiceConnection;
+
+    private ArrayList<Track.Id> trackIds;
+
+    private final Runnable bindChangedStatusCallback = new Runnable() {
```

Comment:
We should change this interface as the ServiceConnection should only call this when it get's connected.
Also in the other activities this code is not suitable for "service died notifications".
And then we can actually pass the service as a parameter.

Reply:
Okay, I followed the same approach than TrackRecordingService. In fact I loved that approach the first time I saw it :roll_eyes: 
Do you know another approach for this?
I'm going to look into it.

Reply:
Creating an interface and implementing on the activity/fragment and pass it to the TrackRecordingService.
(side note: I am working on making TrackController a fragment and might need another callback there aka service disconnected).